compress/font_subsetting.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Font subsetting example by using creator `EnableFontSubsetting` setting
+ * and using optimizer option of `SubsetFonts`.
+ *
+ * Run as: go run font_subsetting.go <font.ttf>
+ */
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/creator"
+	"github.com/unidoc/unipdf/v3/model"
+	"github.com/unidoc/unipdf/v3/model/optimize"
+)
+
+const licenseKey = `
+-----BEGIN UNIDOC LICENSE KEY-----
+Free trial license keys are available at: https://unidoc.io/
+-----END UNIDOC LICENSE KEY-----
+`
+
+const usage = "Usage: %s FONT_TTF_FILE\n"
+const outputCreatorFile = "font_subsetting_creator.pdf"
```

Comment:
Would be better to create two separate examples, the one for the creator can be in the report folder and the one for the optimizer can be here.
Using the creator subsetting should normally be faster, the other one is more generic and attempts to subset all fonts in the document.

Reply:
Actually, there's an example of using `creator.EnableFontSubsetting` in [text/pdf_using_cjk_font.go](https://github.com/unidoc/unipdf-examples/blob/255232026d72b03de23e98c8c22fd28210222849/text/pdf_using_cjk_font.go#L53).

Should I create another one in report folder?